aws-s3-transfer-manager/src/operation/download/service.rs

```diff
@@ -31,6 +32,19 @@ pub(super) struct DownloadChunkRequest {
     pub(super) start_seq: u64,
 }
 
+impl ProvidePayloadSize for DownloadChunkRequest {
+    fn payload_size(&self) -> u64 {
+        // we can't know the actual size by calling next_seq() as that would modify the
+        // state. Instead we give an estimate based on the current sequence.
+        let seq = self.ctx.current_seq();
+        let remaining = self.remaining.clone();
+        let part_size = self.ctx.handle.download_part_size_bytes();
+        let start = remaining.start() + ((seq - self.start_seq) * part_size);
+        let end_inclusive = cmp::min(start + part_size - 1, *remaining.end());
```

Comment:
trivial: can we get a common helper for this math? It's duplicated in the `next_chunk()` function right below here.